script/model.py

```diff
@@ -4,141 +4,350 @@
+      binarized MNIST images
+
+    """
+
+    def __init__(self, x_size, processed_x_size, b_size, z_size):
         super(TD_VAE, self).__init__()
-        self.input_size = input_size
-        self.belief_state_size = belief_state_size
-
-        ## LSTM for aggregating belief states
-        self.lstm = nn.LSTM(input_size = self.input_size,
-                            hidden_size = self.belief_state_size,
-                            batch_first = True)
-
-        ## belief state to state
-        self.state_size = state_size
-        self.belief_to_state_mean = nn.Sequential(
-            nn.Linear(self.belief_state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-
-        self.belief_to_state_logstd = nn.Sequential(
-            nn.Linear(self.belief_state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))               
-        
-        ## state to observation
-        self.state_to_obs = nn.Sequential(
-            nn.Linear(self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.input_size),
-            nn.Sigmoid())        
-
-        ## state transition
-        self.state_to_state_mean = nn.Sequential(
-            nn.Linear(self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-        
-        self.state_to_state_logstd = nn.Sequential(
-            nn.Linear(self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-
-        ## inference state
-        self.infer_state_mu = nn.Sequential(
-            nn.Linear(self.belief_state_size*2 + self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-
-        self.infer_state_logstd = nn.Sequential(
-            nn.Linear(self.belief_state_size*2 + self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))        
-        
+        self.x_size = x_size
+        self.processed_x_size = processed_x_size
+        self.b_size = b_size
+        self.z_size = z_size
+
+        # input pre-processing layer
+        self.process_x = PreProcess(self.x_size, self.processed_x_size)
+
+        # one layer LSTM for aggregating belief states
+        # One layer LSTM is used here and I am not sure how many layers
+        # are used in the original paper
+
+        # batch_first – If True, then the input and output tensors
+        # are provided as (batch, seq, feature) instead of (seq, batch,
+        # feature)
+        self.lstm = nn.LSTM(input_size=self.processed_x_size,
+                            hidden_size=self.b_size,
+                            batch_first=True)
+
+        # Two layer state model is used. Sampling is done by sampling
+        # higher layer first.
```

Comment:
```suggestion
        # The model uses two layers and samples the higher layer first.
```

Please try to use active voice instead of passive voice where possible. Also, I'm not sure what "higher layer first" means 🤔 can you be a bit more specific?

Reply:
This is not my comment so I am not sure what higher layer refers to. I will remove it